optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoAnnotationWrappers.java

```diff
@@ -0,0 +1,740 @@
+                                    break;
+                                case ARRAY:
+                                default:
+                                    throw new IllegalStateException("Arrays of " + value.componentKind() +
+                                            " are unsupported in the enum parser." +
+                                            " This exception was caused by a" +
+                                            " @" + annotationInstance.name() + " annotation" +
+                                            " for parameter " + value.name() + "." +
+                                            " Maybe put " + annotationInstance.name() +
+                                            " not in a subpackage of org.optaplanner?");
+                            }
+                            break;
+                        case NESTED:
+                            AnnotationWrapper wrapper = new AnnotationWrapper(value.asNested(), indexView);
+                            annotationValues.put(value.name(), wrapper.get());
+                            break;
+                        case UNKNOWN:
+                        default:
+                            throw new IllegalStateException(value.componentKind() +
+                                    " are unsupported in the enum parser." +
+                                    " This exception was caused by a" +
+                                    " @" + annotationInstance.name() + " annotation" +
+                                    " for parameter " + value.name() + "." +
+                                    " Maybe put " + annotationInstance.name() +
+                                    " not in a subpackage of org.optaplanner?");
+                    }
+                });
+            }
+        }
+
+        @Override
+        public Annotation get() {
+            if (annotation != null) {
+                return annotation;
+            } else {
+                return AllOptaPlannerAnnotationEnum.getForClass(annotationType, annotationValues);
+            }
+        }
+
+        public Class<? extends Annotation> getAnnotationType() {
+            return annotationType;
+        }
+
+        public void setAnnotationType(Class<? extends Annotation> annotationType) {
+            this.annotationType = annotationType;
+        }
+
+        public Map<String, Object> getAnnotationValues() {
+            return annotationValues;
+        }
+
+        public void setAnnotationValues(Map<String, Object> annotationValues) {
+            this.annotationValues = annotationValues;
+        }
+
+        public Annotation getAnnotation() {
+            return annotation;
+        }
+
+        public void setAnnotation(Annotation annotation) {
+            this.annotation = annotation;
+        }
+    }
+
+    public static class ConstraintConfigurationWrapper implements ConstraintConfiguration {
+        public Map<String, Object> map;
```

Comment:
I debated this, and wonder about how clear it would be map was a public variable from its superclass. Perhaps I can replace using the map with a generic method